### ISSUE TITLE
Implement deactivate actor method

### ISSUE THREAD

**Issue by akkie** [dapr/dotnet-sdk#734]
The actor API allows it to [deactivate an actor](https://docs.dapr.io/reference/api/actors_api/#deactivate-actor). The [testing code](https://github.com/dapr/dotnet-sdk/blob/9104c6624cc474e52717ddb7408a75449b071a02/test/Dapr.Actors.AspNetCore.IntegrationTest/ActivationTests.cs#L52) also has implemented an API call to deactivate an actor. It would be great if the SDK could provide a method in the `ActorProxy` class which allows it to deactivate an actor without to implementing an API call by itself.

**Comment by fabistb** [comment]
@rynowak, can you please add his to our project?
Do you have any thoughts on this topic?

**Comment by halspang** [comment]
@akkie/@fabistb - What are your use cases for this API? Technically, this is an API that Dapr calls into your application, so it isn't something that we would put in the ActorProxy as that's designed to call out to Dapr.

**Comment by halspang** [comment]
Is this something you could tune with the existing actor configuration? You can set the `actorIdleTimeout` to be arbitrarily low. It won't deactivate the actor while it is running but will do so after it stops. Does this solve what you want?

If not, and you'd really like to call the API directly. I'd recommend just making a call out to your own application as Dapr would. My only real concern with that, and putting it in the proxy, is that since we aren't going through Dapr, the runtime could have an improper reference to your actors. In theory, this shouldn't cause any problems as the actor APIs tend to activate the actor with every call, but I don't like the idea of their being a split-state.

**Comment by fabistb** [comment]
From my understanding the downside with the `actorIdleTimeout` setting is that it applies for all actor types within the service.
However we only want to free the resources for one specific actor type and not change the behaviour for the other actor types.

**Comment by halspang** [comment]
Yeah, that is the problem with the actor configuration. Though potentially that's a feature to look into for core Dapr? The ability to tie the configurations to specific types?

As long as the Dapr runtime doesn't expose actor deactivation as an API, I don't think it's something we should put into the `ActorProxy`. If this is a feature you need, I think we should put it into the runtime first, and then we can expose it in the SDK. What do you think, @rynowak?

**Comment by yaron2** [comment]
@halspang we do have this in the runtime: https://docs.dapr.io/reference/api/actors_api/#deactivate-actor.

We should add this method to the SDK IMO.

**Comment by halspang** [comment]
@yaron2 - That's the runtime calling your application, not the other way around. Dapr does not expose that endpoint in the runtime. https://github.com/dapr/dapr/blob/a02eb997c6399939148aa483c644613dcacc3056/pkg/http/api.go#L272

If we want something like the `ActorProxy` to handle this, I think it should be an API that the dapr runtime vends. As an alternate, I'm not opposed to exposing an internal cleanup method in the SDK outside of the proxy though. However, similar to activating an Actor, I think we should go through the runtime and would be happy to expose that API in the runtime.

**Comment by rynowak** [comment]
@yaron2 - what @halspang  is saying this is exactly the conclusion that I came to. 

There is no contract provided by the **runtime** for a *client to deactivate an actor*.  If this is a desirable thing it should be part of the surface area for the client.

**Comment by AntonSmolkov** [comment]
*+* for the runtime supported deactivation method.
Now it's not possible to immediately stop the actor if it has messages in it's queue.

i.e. current implementation works more like `PoisonPill` from Akka.Net, not like regular `Stop()` method.

https://petabridge.com/blog/how-to-stop-an-actor-akkadotnet/
